modules/hostap/src/supp_main.c

```diff
@@ -306,9 +306,32 @@ static void iface_event_handler(struct net_mgmt_event_callback *cb,
 							uint32_t mgmt_event, struct net_if *iface)
 {
 	const char *ifname = iface->if_dev->dev->name;
+	char *tmp_csv;
+	char *prefix;
+	char *mgd_if_prefix = CONFIG_WPA_SUPP_MGD_IFACES_PREFIX;
+	bool found = false;
+
+	if (strlen(mgd_if_prefix) > 0) {
+		tmp_csv = os_zalloc(strlen(mgd_if_prefix));
+		if (!tmp_csv) {
+			wpa_printf(MSG_ERROR, "Failed to allocate memory for prefix string");
+			return;
+		}
+		strcpy(tmp_csv, mgd_if_prefix);
+		prefix = strtok(tmp_csv, ",");
 
-	if (strncmp(ifname, IFACE_MATCHING_PREFIX, sizeof(IFACE_MATCHING_PREFIX) - 1) != 0) {
-		return;
+		while (prefix) {
+			if (strncmp(ifname, prefix, strlen(prefix)) == 0) {
```

Comment:
We can either write '\0` where the `,` is or use proper len (as we know the start and stop of the prefix) when doing the compare.

Reply:
Will use `strtok` which returns a null-terminated string.